Neos.Neos/Classes/Domain/Service/FusionService.php

```diff
@@ -136,57 +137,40 @@ public function createRuntime(TraversableNodeInterface $currentSiteNode, Control
     public function getMergedFusionObjectTree(TraversableNodeInterface $startNode)
     {
         $siteResourcesPackageKey = $this->getSiteForSiteNode($startNode)->getSiteResourcesPackageKey();
-
         $siteRootFusionPathAndFilename = sprintf($this->siteRootFusionPattern, $siteResourcesPackageKey);
-        $siteRootFusionCode = $this->readExternalFusionFile($siteRootFusionPathAndFilename);
-
-        $mergedFusionCode = '';
-        $mergedFusionCode .= $this->generateNodeTypeDefinitions();
-        $mergedFusionCode .= $this->getFusionIncludes($this->prepareAutoIncludeFusion());
-        $mergedFusionCode .= $this->getFusionIncludes($this->prependFusionIncludes);
-        $mergedFusionCode .= $siteRootFusionCode;
-        $mergedFusionCode .= $this->getFusionIncludes($this->appendFusionIncludes);
 
-        return $this->fusionParser->parse($mergedFusionCode, $siteRootFusionPathAndFilename);
+        return $this->fusionParser->parseFrom(
+            FusionSourceCodeCollection::fromArray(array_filter([
+                ...$this->generateNodeTypeDefinitions(),
+                ...$this->prepareAutoIncludeFusion(),
```

Comment:
The autoincludes are now separate new entry points instead of concatenated fusion code with $siteRootFusionPathAndFilename as wrong context file 

fixes
https://github.com/neos/neos-development-collection/issues/3835